lib/sqlalchemy/sql/elements.py

```diff
@@ -3869,7 +3955,7 @@ def __init__(self, start, stop, step, _name=None):
         )
         self.type = type_api.NULLTYPE
 
-    def self_group(self, against=None):
+    def self_group(self, against: Optional[OperatorType] = None) -> Slice:
         assert against is operator.getitem
```

Comment:
Getting a non-overlapping identity check in the assert. `operator.getitem()` is interpreted as an overloaded function by mypy. Hovewer the overloads it sees are from `typeshed-fallback/stdlib/_operator.pyi` stub. Guess it shouldn't work that way